extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/utils/PaginationImplementor.java

```diff
@@ -119,7 +119,12 @@ private ResultHandle getPageUri(BytecodeCreator creator, ResultHandle uriInfo, R
      */
     private ResultHandle getLastPage(BytecodeCreator creator, ResultHandle page, ResultHandle pageCount) {
         ResultHandle pageNumber = creator.invokeStaticMethod(
-                ofMethod(Integer.class, "sum", int.class, int.class, int.class), pageCount, creator.load(-1));
+                ofMethod(Math.class, "max", int.class, int.class, int.class),
+                creator.invokeStaticMethod(
+                        ofMethod(Integer.class, "sum", int.class, int.class, int.class),
+                        pageCount,
+                        creator.load(-1)),
```

Comment:
Can we replace this snippet with the code below?

```suggestion
                creator.add(pageCount, creator.load(-1)),
```

Reply:
Maybe. But frankly, I prefer avoid touching it more than necessary.